test/hooks.test.js

```diff
@@ -259,6 +259,98 @@ test('preHandler hook should support encapsulation / 5', t => {
   })
 })
 
+test('onRoute hook should be called / 1', t => {
+  t.plan(2)
+  const fastify = Fastify()
+
+  fastify.register((instance, opts, next) => {
+    instance.addHook('onRoute', () => {
+      t.pass()
+    })
+    instance.get('/', opts, function (req, reply) {
+      reply.send()
+    })
+    next()
+  })
+
+  fastify.ready(err => {
+    t.error(err)
+  })
```

Comment:
In this test you are not checking the order.

Reply:
The tests are copied from `onClose` tests

Reply:
I don't have to I only have to ensure that `onRoute` handler is called once. The circumstance that `ready` is called only once belongs to a different test.

Reply:
Truly, the order is very important. this test isn't check it. So, could you add a test for checking the order?

Reply:
the test "onRoute hook should be called / 2" does it